### PR TITLE
Skip Obj-C code generation if output folder was not provided

### PR DESCRIPTION
**Pull request by freitass**
Fixes the following exception when issuing a Djinni command to generate code other than Obj-C:
```Exception in thread "main" java.util.NoSuchElementException: None.get
        at scala.None$.get(Option.scala:529)
        at scala.None$.get(Option.scala:527)
        at djinni.ObjcGenerator.writeObjcFile(ObjcGenerator.scala:412)
        at djinni.ObjcGenerator.generateInterface(ObjcGenerator.scala:104)
        at djinni.Generator.$anonfun$generate$2(generator.scala:363)
        at djinni.Generator.$anonfun$generate$2$adapted(generator.scala:358)
        at scala.collection.LinearSeqOptimized.foreach(LinearSeqOptimized.scala:75)
        at scala.collection.LinearSeqOptimized.foreach$(LinearSeqOptimized.scala:72)
        at scala.collection.mutable.MutableList.foreach(MutableList.scala:33)
        at djinni.Generator.generate(generator.scala:358)
        at djinni.generatorTools.package$.generate(generator.scala:222)
        at djinni.Main$.main(Main.scala:374)
        at djinni.Main.main(Main.scala)
```

### CODE REVIEW

**Review by a4z — COMMENTED** [review]
consistent with the rest, thanks! 

since you are looking at that code, 
could you please also check, when `--skip-generation` is used, if the part below your changes works
`if (spec.objcSwiftBridgingHeaderWriter.isDefined) { ...` 

because reading the code seems this file will always be written if `--skip-generation` is set, but a `--objc-swift-bridging-header` 

(maybe the whole thing is a bit overcomplicated, but for all other parts, `--skip-generation` will also disable code gen even if output locations are specified, I think

File: src/main/scala/djinni/generator.scala [hunk 1]
```diff
@@ -116,9 +116,15 @@ package object generatorTools {
     val underCaps = (s: String) => s.toUpperCase
     val prefix = (prefix: String, suffix: IdentConverter) => (s: String) => prefix + suffix(s)
 
-    val javaDefault = JavaIdentStyle(camelUpper, camelUpper, camelLower, camelLower, camelLower, underCaps, underCaps)
-    val cppDefault = CppIdentStyle(camelUpper, camelUpper, camelUpper, underLower, underLower, underLower, underCaps, underCaps)
-    val objcDefault = ObjcIdentStyle(camelUpper, camelUpper, camelLower, camelLower, camelLower, camelUpper, camelUpper)
+    val javaDefault = JavaIdentStyle(ty = camelUpper, typeParam = camelUpper,
```
[the remark this answers is not in the document]
Comment (freitass) [reply src/main/scala/djinni/generator.scala:119]: Oh, sorry! Yes we did, I'll revert it.

[the remark this answers is not in the document]
Comment (a4z) [reply src/main/scala/djinni/generator.scala]: Thanks

**Comment by freitass** [comment]
> because reading the code seems this file will always be written if `--skip-generation` is set, but a `--objc-swift-bridging-header`

You're right. The Swift bridging headers will be written if `--objc-swift-bridging-header`, regardless of `--skip-generation`. In fact, there's a `createFolder(...)` call in https://github.com/cross-language-cpp/djinni-generator/blob/c2f2fcd657b7adfad1bc97309991af00eef23d85/src/main/scala/djinni/Main.scala#L313 which is not consistent with the other languages.

The other two occurrences of `createFolder(...)` in `Main.scala` are for the input and output files list. I believe it'd make more sense for the Swift logic to be moved to `generator.scala`, but since I have never used the Swift code gen I'm afraid of making those changes without being able to test it.

**Comment by a4z** [comment]
I was actually referring to that:
https://github.com/cross-language-cpp/djinni-generator/blob/c2f2fcd657b7adfad1bc97309991af00eef23d85/src/main/scala/djinni/generator.scala#L230

that maybe needs a `if (!spec.skipGeneration)` above, like the others

about the folder generation, we could put that into an extra ticket

**Comment by freitass** [comment]
I agree with you, this change might be a little bit more involved. What I meant is that protecting that block with an `if` wouldn't solve the problem completely, and given how uncomfortable I currently am to make that change I'd go with a separate ticket.

**Comment by a4z** [comment]
I added a suggestion, to skip bridge header generation if the skip code gen is provided.
Please check @freitass

**Review by freitass — COMMENTED** [review]
Thanks for submitting the code!

**Review by jothepro — APPROVED** [review]
I'm going to add some tests for this edge-case ASAP.
